File: hw/palomo/u3/hw25TemplateMethod/TemplateMethod.py/ec_edu_espe/templatemethod/controller/beverage_test.py
"""
@author Cristian Palomo, Error 404, @ESPE
"""
from ec_edu_espe.templatemethod.model.coffee import Coffee
from ec_edu_espe.templatemethod.model.tea import Tea
from ec_edu_espe.templatemethod.view.beverage_view import BeverageView


def main() -> None:
    view = BeverageView()

    tea = Tea()
    coffee = Coffee()

    view.print_message("\nMaking tea ...")
    tea.prepare_recipe()

    view.print_message("\nMaking coffee ...")
    coffee.prepare_recipe()


if __name__ == "__main__":
    main()

File: hw/palomo/u3/hw25TemplateMethod/TemplateMethod.js/src/controller/BeverageTest.js
/**
 * @author Cristian Palomo, Error 404, @ESPE
 */
import { Coffee } from "../model/Coffee.js";
import { Tea } from "../model/Tea.js";
import { BeverageView } from "../view/BeverageView.js";

async function main() {
    const view = new BeverageView();

    const tea = new Tea();
    const coffee = new Coffee();

    view.printMessage("\nMaking tea ...");
    await tea.prepareRecipe();

    view.printMessage("\nMaking coffee ...");
    await coffee.prepareRecipe();
}

main();
